.changeset/honest-apricots-fail.md

```diff
@@ -0,0 +1,68 @@
+---
+'@shopify/cli-hydrogen': patch
+---
+
+# New `h2 upgrade` command
+
+We are introducing a new hydrogen cli `upgrade` command that:
+
+- Makes upgrading hydrogen as easy as typing `h2 upgrade` in your terminal
+- Provides a quick summary of the key `features` and `fixes` of any available
+  hydrogen version(s)
+- Generates a `TODO` instructions file detailing all cumulative code changes required
+  to adopt a given hydrogen version
+- Provides a gentle notice during development detailing when a hydrogen is outdated, as well as a quick glance into the number of hydrogen version available
+
+## Basic use
+
+```bash
+# from the base of the project run
+
+h2 upgrade
+```
+
+Alternatively, you can upgrade from another folder by using the `--path` flag
+
+```bash
+h2 upgrade --path /path-to-a-outdated-hydrogen-project
```

Comment:
[Just commenting here for threading]

When tophatting I get this error when running the `dev` command in the CLI package, something to do with missing mini-oxygen?

![image](https://github.com/Shopify/hydrogen/assets/547470/f7606984-03fe-423f-b8e9-5c975d74d220)

Full error text:
<details>
<pre>
Error: Command failed: node ./scripts/generate-manifest.mjs
/Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@oclif/core/lib/module-loader.js:80
                throw new errors_1.ModuleLoadError(`${isESM ? 'import()' : 'require'} failed to load ${filePath || modulePath}: ${error.message}`);
                      ^

ModuleLoadError Plugin: @shopify/cli-hydrogen: [MODULE_NOT_FOUND] import() failed to load /Users/gfs/src/github.com/Shopify/hydrogen/packages/cli/dist/commands/hydrogen/check.js: Cannot find package '/Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@shopify/mini-oxygen/package.json' imported from /Users/gfs/src/github.com/Shopify/hydrogen/packages/cli/dist/lib/request-events.js
    at ModuleLoader.loadWithData (/Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@oclif/core/lib/module-loader.js:80:23)
    at async fetch (/Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@oclif/core/lib/config/plugin.js:192:53)
    at async Plugin.findCommand (/Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@oclif/core/lib/config/plugin.js:208:21)
    at async /Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@oclif/core/lib/config/plugin.js:251:62
    at async Promise.all (index 1)
    at async Plugin._manifest (/Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@oclif/core/lib/config/plugin.js:249:24)
    at async Plugin.load (/Users/gfs/src/github.com/Shopify/hydrogen/node_modules/@oclif/core/lib/config/plugin.js:135:25)
    at async file:///Users/gfs/src/github.com/Shopify/hydrogen/packages/cli/scripts/generate-manifest.mjs:33:1 {
  oclif: { exit: 1 },
  code: 'MODULE_NOT_FOUND',
  detail: 'module: @oclif/core@2.11.7\n' +
    'task: toCached\n' +
    'plugin: @shopify/cli-hydrogen\n' +
    'root: /Users/gfs/src/github.com/Shopify/hydrogen/packages/cli\n' +
    'See more details with DEBUG=*'
}

Node.js v20.9.0

    at ChildProcess.exithandler (node:child_process:422:12)
    at ChildProcess.emit (node:events:514:28)
    at maybeClose (node:internal/child_process:1105:16)
    at Socket.<anonymous> (node:internal/child_process:457:11)
    at Socket.emit (node:events:514:28)
    at Pipe.<anonymous> (node:net:337:12)
</pre>
</details>

I tried running npm install, different Node versions, adding mini-oxygen to the package file, but no dice.